### PR TITLE
Fix toggle buttons and other multi-dimension interactable themes 

### PR DESCRIPTION
**Pull request by julenka**
## Overview
Prior to #6104 Interactable would create all themes for all dimensions at startup, and then just select the appropriate list for the current dimension.

#6104  accidentally re-wrote that to recreate the list of current themes every time the dimension changed. When a theme is created and inits, it looks at the current offset/position/rotation etc as the starting position from which to base offset. In case of #6178 and #6187 the buttons had not reset to their original position before the new theme is created, causing new themes to be based not on the original 'resting' transform, but on some transform in middle of animation. This caused things like toggle buttons to move forward steadily as toggle was triggered.

Big thank you to @ritijain and @gilbdev for finding these issues!

## Changes
- Fixes: #6178
- Fix this by reverting to old behavior -- creating list of all themes per dimension whenever interactable is created, or when profiles change. 
- Add test to catch this regression in future, thank you @cdiaz-ms for helping write the test.

## Verification
Manually verified toggles behave correctly
Verified that test did not pass before changes, and that the test passes after changes.

> This optional section is a place where you can detail the specific type of verification 
> you want from reviewers. For example, if you want reviewers to checkout the PR locally
> and validate the functionality of specific scenarios, provide instructions
> on the specific scenarios and what you want verified.
>
> If there are specific areas of concern or question feel free to highlight them here so
> that reviewers can watch out for those issues.
>
> As a reviewer, it is possible to check out this change locally by using the following
> commands (substituting {PR_ID} with the ID of this pull request):
>
> git fetch origin pull/{PR_ID}/head:name_of_local_branch
>
> git checkout name_of_local_branch

### CODE REVIEW

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs [hunk 1]
```diff
@@ -576,6 +578,52 @@ public IEnumerator TestInteractableDisableOnClick()
...
+
+            int numClicks = 3;
+            for (int i = 0; i < numClicks; i++)
+            {
+                wasClicked = false;
+                yield return rightHand.Click();
+                // Wait for button animation to complete
+                yield return new WaitForSeconds(0.33f);
+
+                Assert.True(wasClicked, "Toggle button was not clicked");
```
Comment (ritijain) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs]: >Assert.True(wasClicked, "Toggle button was not clicked"); [](start = 16, length = 57)

Can you also add validation for CurrentDimension ? (0 on 1st click, 1 on second click and 0 again on 3rd click )

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs [hunk 2]
```diff
@@ -576,6 +578,52 @@ public IEnumerator TestInteractableDisableOnClick()
...
+
+            Assert.True(interactable.IsEnabled);
+
+            bool wasClicked = false;
+            interactable.OnClick.AddListener(() => { wasClicked = true; });
+
+            // Get start position of the front plate before button is pressed
+            Vector3 frontPlateStartPosition = frontPlateTransform.localPosition;
+
+            yield return rightHand.Show(p2);
```
Comment (ritijain) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs]: >   yield return rightHand.Show(p2); [](start = 9, length = 35)

should this result in "interactable.HasFocus == true" ? if so, an assert for that would be helpful here.

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs (hunk 1 above)
Comment (julenka) [reply Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs]: That's covered in TestDimensions, not sure it's needed here?

**Pull request merged**